pkg/controller/account/iam.go

```diff
@@ -731,3 +731,207 @@ func (r *ReconcileAccount) createManagedOpenShiftSupportRole(reqLogger logr.Logg
 
 	return roleID, err
 }
+
+func (r *ReconcileAccount) ProbeSecret(reqLogger logr.Logger, currentAcctInstance *awsv1alpha1.Account,
+	secretNamespacedName types.NamespacedName, awsAssumedRoleClient awsclient.Client, iamUserUHC string, nameSpace string) error {
+
+	var secretName *string
+	iamUserSecretName := createIAMUserSecretName(currentAcctInstance.Name)
+	secretName = &iamUserSecretName
+	kubeSecretNamespacedName := types.NamespacedName{Name: iamUserSecretName, Namespace: nameSpace}
+	// Check if secret exist in Kubernetes
+	secretExists, err := r.DoesSecretExist(kubeSecretNamespacedName)
+	if err != nil {
+		reqLogger.Error(err, fmt.Sprintf("Unable check if secret: %s exists", secretNamespacedName.String()))
+		return err
+	}
+
+	if !secretExists {
+		// If secret doesn't exist, create new one
+		secretName, err = r.BuildIAMUser(reqLogger, awsAssumedRoleClient, currentAcctInstance, iamUserUHC, nameSpace)
+		if err != nil {
+			reason, errType := getBuildIAMUserErrorReason(err)
+			errMsg := fmt.Sprintf("Failed to recreate IAM UHC user %s: %s", iamUserUHC, err)
+			_, stateErr := r.setAccountFailed(
+				reqLogger,
+				currentAcctInstance,
+				errType,
+				reason,
+				errMsg,
+				AccountFailed,
+			)
+			if stateErr != nil {
+				reqLogger.Error(err, "failed setting account state", "desiredState", AccountFailed)
+			}
+			return err
+		}
+	} else {
+		// If secret exists, check that credentials are valid
+		validSecret, err := r.IsKubeSecretValid(reqLogger, currentAcctInstance)
+		if err != nil {
+			reqLogger.Error(err, "failed secret validation")
+		}
+		if !validSecret {
+			// If credentials aren't valid, make them valid again
+			secretName, err = r.RevalidateSecret(reqLogger, awsAssumedRoleClient, currentAcctInstance, iamUserUHC, nameSpace)
+			if err != nil {
+				reason, errType := getBuildIAMUserErrorReason(err)
+				errMsg := fmt.Sprintf("Failed to revalidate IAM UHC user %s: %s", iamUserUHC, err)
+				_, stateErr := r.setAccountFailed(
+					reqLogger,
+					currentAcctInstance,
+					errType,
+					reason,
+					errMsg,
+					AccountFailed,
+				)
+				if stateErr != nil {
+					reqLogger.Error(err, "failed setting account state", "desiredState", AccountFailed)
+				}
+				return err
+			}
+		}
+	}
+
+	currentAcctInstance.Spec.IAMUserSecret = *secretName
+	err = r.accountSpecUpdate(reqLogger, currentAcctInstance)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *ReconcileAccount) IsKubeSecretValid(reqLogger logr.Logger, currentAcctInstance *awsv1alpha1.Account) (bool, error) {
+
+	// Build new aws client with credentials inside secret
+	awsClient, err := r.awsClientBuilder.GetClient(controllerName, r.Client, awsclient.NewAwsClientInput{
+		SecretName: currentAcctInstance.Spec.IAMUserSecret,
+		NameSpace:  currentAcctInstance.Namespace,
+		AwsRegion:  "us-east-1",
```

Comment:
Is it a possible scenario that iam credentials are valid in another region than `us-east-1`? Or do we require the credentials to always be valid in `us-east-1`?

Reply:
aws creds are valid globally as far as i know. Even the temporarily ones